pkg/sql/alter_type.go

```diff
@@ -191,12 +200,53 @@ func (p *planner) renameTypeValue(
 	n.desc.EnumMembers[enumMemberIndex].LogicalRepresentation = newVal
 
 	return p.writeTypeChange(
-		params.ctx,
+		ctx,
 		n.desc,
-		tree.AsStringWithFQNames(n.n, params.Ann()),
+		jobDesc,
 	)
 }
 
+func (p *planner) setTypeSchema(
+	ctx context.Context,
+	n *alterTypeNode,
+	schema string,
+	jobDesc string,
+) error {
+	typeDesc := n.desc
+	schemaID := typeDesc.GetParentSchemaID()
+	databaseID := typeDesc.GetParentID()
+
+	desiredSchemaID, err := p.prepareSetSchema(ctx, typeDesc, schema)
+	if err != nil {
+		return err
+	}
+
+	// If the schema being changed to is the same as the current schema for the
+	// type, do a no-op.
+	if desiredSchemaID == schemaID {
+		return nil
+	}
+
+	typeDesc.AddDrainingName(databaseID, schemaID, typeDesc.Name)
+
+	// Set the tableDesc's new schema id to the desired schema's id.
+	typeDesc.SetParentSchemaID(desiredSchemaID)
+
+	if err := p.writeTypeChange(ctx, typeDesc, jobDesc); err != nil {
+		return err
+	}
+
+	newKey := sqlbase.MakeObjectNameKey(ctx, p.ExecCfg().Settings,
+		databaseID, desiredSchemaID, typeDesc.Name).Key(p.ExecCfg().Codec)
+
+	b := &kv.Batch{}
```

Comment:
Nit -- can we pull this into a function like "WriteNameKey" that takes in a key, writes it to a batch and does the kv tracing check? I feel like i've seen this repeated so many times